languages/zh-tw/constants.genie

```diff
@@ -0,0 +1,134 @@
+    '這 禮拜 以前' => new Ast.Value.Date(Ast.DateEdge('end_of', 'week'), '+', null);
+    '上 星期'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '-', Ast.Value.Measure(1, 'week'));
+    '上週'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '-', Ast.Value.Measure(1, 'week'));
+    '上 禮拜'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '-', Ast.Value.Measure(1, 'week'));
+    '這個 月'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'mon'), '+', null);
+    '今年'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'year'), '+', null);
+    '下個 月'           => new Ast.Value.Date(Ast.DateEdge('end_of', 'mon'), '+', null);
+    '明年'             => new Ast.Value.Date(Ast.DateEdge('end_of', 'year'), '+', null);
+    '上個 月'           => new Ast.Value.Date(Ast.DateEdge('end_of', 'mon'), '-', Ast.Value.Measure(1, 'mon'));
+    '去年'             => new Ast.Value.Date(Ast.DateEdge('end_of', 'year'), '-', Ast.Value.Measure(1, 'year'));
+    }
+}
+
+constant_Date = {
+    !turking {
+    duration:constant_Measure_ms '以內' => new Ast.Value.Date(null, '+', duration);
+    duration:constant_Measure_ms '之內' => new Ast.Value.Date(null, '+', duration);
+    duration:constant_Measure_ms '以前' => new Ast.Value.Date(null, '-', duration);
+    duration:constant_Measure_ms '之前' => new Ast.Value.Date(null, '-', duration);
+    duration:constant_Measure_ms pt:constant_date_point '以後' => new Ast.Value.Date(pt.value, '+', duration);
+    duration:constant_Measure_ms pt:constant_date_point '之後' => new Ast.Value.Date(pt.value, '+', duration);
+    duration:constant_Measure_ms pt:constant_date_point '以前' => new Ast.Value.Date(pt.value, '-', duration);
+    duration:constant_Measure_ms pt:constant_date_point '之前' => new Ast.Value.Date(pt.value, '-', duration);
+    }
+}
+
+constant_Measure_byte = {
+    !turking num:constant_Number 'byte' => C.addUnit('byte', num);
+    num:constant_Number 'kb' => C.addUnit('KB', num);
+    num:constant_Number 'mb' => C.addUnit('MB', num);
+    num:constant_Number 'gb' => C.addUnit('GB', num);
+    num:constant_Number 'tb' => C.addUnit('TB', num);
+    num:constant_Number 'kilobytes' => C.addUnit('KB', num);
+    num:constant_Number 'megabytes' => C.addUnit('MB', num);
+    num:constant_Number 'gigabytes' => C.addUnit('GB', num);
+    num:constant_Number 'terabytes' => C.addUnit('TB', num);
+}
+
+constant_Measure_kg = {
+    !turking num:constant_Number '公克' => C.addUnit('g', num);
+    num:constant_Number '公斤' => C.addUnit('kg', num);
+    num:constant_Number '磅'   => C.addUnit('lb', num);
+    num:constant_Number '盎司' => C.addUnit('oz', num);
+}
+
+constant_Measure_C = {
+    !turking {
+        num:constant_Number 'c' => C.addUnit('C', num);
+        num:constant_Number '攝氏' => C.addUnit('C', num);
+    }
+    num:constant_Number 'f' => C.addUnit('F', num);
+    num:constant_Number '華氏' => C.addUnit('F', num);
+    num:constant_Number '度' => C.addUnit('TB', num);
```

Comment:
Wrong unit here (and ugh it's broken in English too...)

Also, the English is written from an American point of view (using imperial units) but make sure the Chinese uses metric where ambiguous.

Reply:
OK I am not quite sure what TB is here...

Reply:
if that just means "degrees" it should be the default measure unit for temperatures: C in Chinese and F in (American) English
The TB there it's a bug in the current code

Reply:
Got it. I delete it here.